### PR TITLE
L1T Phase-1: Adding L1boosted objects

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This PR is to include the L1TCaloSummary emulator into the L1TCaloLayer1 package.
The input to the emulator are the regions produced by L1TCaloLayer1: < L1CaloRegionCollection >.
The output of the emulator is the L1boosted object collection of type vector < l1extra::L1JetParticle >.

EDProducer: L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc
Config to run the EDProducer:
L1Trigger/L1TCaloLayer1/python/uct2016EmulatorDigis_cfi.py

NOTE: To run uct2016EmulatorDigis it is now essential to run simCaloStage2Layer1Digis that will produce the regions.

#### PR validation:

Rates and efficiency studies presented to L1 DPG:
https://indico.cern.ch/event/1023711/#8-boosted-object-identificatio

Recipe to run an analyzer for the L1boosted performance:
https://github.com/pallabidas/Run3Ntuplizer/tree/L1boosted_L1TOffline

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Rebase of https://github.com/cms-l1t-offline/cmssw/pull/920 and https://github.com/cms-l1t-offline/cmssw/pull/907

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
+l1

File: L1Trigger/L1TCaloLayer1/src/UCTGeometry.cc [hunk 1]
```diff
@@ -184,10 +184,22 @@ UCTTowerIndex UCTGeometry::getUCTTowerIndex(UCTRegionIndex region, uint32_t iEta
     return UCTTowerIndex(0, 0);  // Illegal values
   }
   int regionEta = region.first;
-  int absRegionEta = std::abs(regionEta);
-  int towerEta = (regionEta / absRegionEta) * (absRegionEta * NEtaInRegion + iEta);
   uint32_t regionPhi = region.second;
-  int towerPhi = regionPhi * NPhiInRegion + iPhi + 1;
+  bool negativeSide = (regionEta < 0);
+  uint32_t regionNo = abs(regionEta) - 1;
```
Comment (qliphy) [inline L1Trigger/L1TCaloLayer1/src/UCTGeometry.cc:189]: abs -> std::abs

File: L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+        if (object->boostedJetRegionET()[3 * iEta + iPhi] > 30 &&
+            object->boostedJetRegionET()[3 * iEta + iPhi] > object->et() * 0.0625)
+          activeStrip = true;
+      }
+      if (activeStrip)
+        activeRegionPhiPattern |= (0x1 << iPhi);
+    }
+    string regionEta = activeRegionEtaPattern.to_string<char, std::string::traits_type, std::string::allocator_type>();
+    string regionPhi = activeRegionPhiPattern.to_string<char, std::string::traits_type, std::string::allocator_type>();
+    if (abs(eta) < 2.5 && (regionEta == "010" || regionPhi == "010" || regionEta == "110" || regionPhi == "110" ||
```
Comment (qliphy) [inline L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc:240]: abs -> std::abs

**Comment by qliphy** [comment]
There are also "Static analyzer inherits from legacy modules warnings", for which you may refer to
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5afc71/16319/llvm-analysis/legacy-mod-sa.txt

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
I have applied the comments, thanks for the review!

**Comment by cecilecaillol** [comment]
+l1

**Comment by qliphy** [comment]
+1

**Pull request merged**